ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -1065,9 +1065,18 @@ - (ORKStepViewController *)viewControllerForStep:(ORKStep *)step {
             
             // Get the step result associated with this step
             ORKStepResult *result = nil;
-            result = _managedResults[step.identifier];
-            if (!result ) {
-                result = [_defaultResultSource stepResultForStepIdentifier:step.identifier];
+            ORKStepResult *previousResult = _managedResults[step.identifier];
+            
+            // Check the default source first
+            BOOL alwaysCheckForDefaultResult = ([self.defaultResultSource respondsToSelector:@selector(alwaysCheckForDefaultResult)] &&
+                                                [self.defaultResultSource alwaysCheckForDefaultResult]);
+            if ((previousResult == nil) || alwaysCheckForDefaultResult) {
+                result = [self.defaultResultSource stepResultForStepIdentifier:step.identifier];
+            }
+            
+            // If nil, assign to the previous result
+            if (!result) {
+                result = previousResult;
             }
             
             if (!result) {
```

Comment:
Please condense down the two (!result) sequential if statements to a more recognized paradigm. 

One possibility is something like this:

if (!result) {
	result = previousResult ? previousResult : [[ORKStepResult alloc] initWithIdentifier:step.identifier];
}